typehints/microbit/microbit/__init__.pyi

```diff
@@ -106,28 +106,21 @@ class MicroBitDigitalPin:
     ``set_pull`` to change the pull mode from the default.
     """
 
-    PULL_UP: int
-    PULL_DOWN: int
-    NO_PULL: int
+    NO_PULL: int = 0
+    PULL_UP: int = 1
+    PULL_DOWN: int = 2
 
     def read_digital(self) -> int:
         """Return 1 if the pin is high, and 0 if it's low."""
 
-    def write_digital(self, value: int) -> None:
-        """Set the pin to high if ``value`` is 1, or to low, if it is 0."""
-
-    def set_pull(self, value: int) -> None:
+    def set_pull(self, value: int = (NO_PULL or PULL_UP or PULL_DOWN)) -> None:
         """Set the pull state to one of three possible values: ``pin.PULL_UP``,
         ``pin.PULL_DOWN`` or ``pin.NO_PULL`` (where ``pin`` is an instance of
         a pin). See below for discussion of default pull states.
         """
 
-
-class MicroBitAnalogDigitalPin(MicroBitDigitalPin):
-    def read_analog(self) -> int:
-        """Read the voltage applied to the pin, and return it as an integer
-        between 0 (meaning 0V) and 1023 (meaning 3.3V).
-        """
+    def write_digital(self, value: int) -> None:
+        """Set the pin to high if ``value`` is 1, or to low, if it is 0."""
 
     def write_analog(self, value: int) -> None:
```

Comment:
Why `MicroBitDigitalPin` contains `*analog*` methods? According to [the docs](http://microbit-micropython.readthedocs.io/en/latest/pin.html?highlight=MicroBitAnalogDigitalPin#microbit.MicroBitDigitalPin), it has only `read_digital` and `write_digital`. The `*analog*` methods are listed in the docs are belonging to `MicroBitAnalogDigitalPin`.

Reply:
I have been in contact with the micro:bit folks and opened this [issue](https://github.com/bbcmicrobit/micropython/issues/487). I was contacted by them to modify the documentation and did with this [pull request](https://github.com/bbcmicrobit/micropython/pull/488).

The issue is how one defines digital vs analog. All micro:bit digital pins have the capability to operate in a PWM mode, which is called "analog_write". Analog pins have the capability to perform a read_analog. The current documentation does not reflect this and it also does not reflect the ability to set pull resistor type. By moving the PWM and pull methods into the digital pin class, this resolves the issue. 

So based on the current definitions, any pin that is not an analog pin, cannot perform PWM operations.

If you look at [Edge Connector Data Sheet](https://github.com/bbcmicrobit/micropython/pull/488) and the [run-time documentation](https://lancaster-university.github.io/microbit-docs/ubit/io/), it may provide a better illustration of what I am saying.

By the way, even though there are 6 analog (input) pins, only one may be used at a time because a single A/D converter is multiplexed across all 6 pins. In addition, only 3 digital pins may be used in PWM mode simultaneously due to limitations of the DAL DynamicPWM driver

Reply:
Thank you for the clarification!